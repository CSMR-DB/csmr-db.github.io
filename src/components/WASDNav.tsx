import React from 'react'
import styled, { StyledComponent } from 'styled-components'
import { IRouteObject } from '../data/routes'
import { theme } from '../data/theme'
import { Link } from 'gatsby'
import {
  ColorProperty,
  BorderProperty,
  GridAreaProperty,
  GridGapProperty,
  BackgroundProperty,
  BackgroundColorProperty,
} from 'csstype'
import { Tween } from 'react-gsap'

const WASDGrid: StyledComponent<
  'nav',
  any,
  { gap: GridGapProperty<1> },
  never
> = styled.nav`
  display: grid;
  justify-content: space-between;
  align-items: center;
  grid-template-areas:
    '. w .'
    'a s d';
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  grid-gap: ${({ gap }: { gap: GridGapProperty<1> }): GridGapProperty<1> =>
    gap};

  & > * {
    justify-self: center;
  }

  @media ${theme.breakpoints.max.smartphone} {
    grid-gap: 2rem;
  }
`

interface IStyledLinkOrbProps {
  color: ColorProperty
  bordercolor: ColorProperty
  area: GridAreaProperty
  background?: BackgroundProperty<1>
  inverted?: boolean
}

const StyledLinkOrb: StyledComponent<
  typeof Link,
  any,
  IStyledLinkOrbProps,
  never
> = styled(Link)`
  height: 8rem;
  width: 8rem;
  margin: 0 auto;
  text-align: center;
  text-decoration: none;
  background: ${({
    background = 'transparent',
  }: IStyledLinkOrbProps): BackgroundProperty<1> => background};
  border: ${({ bordercolor }: IStyledLinkOrbProps): BorderProperty<1> =>
    `1px solid ${bordercolor}`};
  border-radius: 256rem;
  color: ${({ color }: IStyledLinkOrbProps): ColorProperty => color};
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  grid-area: ${({ area }: IStyledLinkOrbProps): GridAreaProperty => area};

  &:hover {
    border: 1px solid ${theme.primaryHover};
    color: ${theme.primaryHover};
  }

  &.active {
    border: 1px solid ${theme.primary};
    color: ${theme.primary};
  }

  @media ${theme.breakpoints.max.smartphone} {
    height: 4rem;
    width: 4rem;
  }
`

const StyledLinkOrbTitle: StyledComponent<'h1', any, {}, never> = styled.h1`
  line-height: inherit;
  margin: 0;

  @media ${theme.breakpoints.max.smartphone} {
    font-size: 0.75rem;
  }
`

const StyledLinkOrbKey: StyledComponent<'span', any, {}, never> = styled.span`
  border: 1px solid #ddd;
  border-radius: 4px;
  width: 2rem;
  height: 2rem;
  margin: 0 auto;
  line-height: 2rem;

  @media ${theme.breakpoints.max.smartphone} {
    display: none;
  }
`

interface IWASDNavProps {
  routes: IRouteObject[]
  color?: ColorProperty
  gap: GridGapProperty<1>
  background: BackgroundColorProperty
  isLandingPage: boolean
}

export function WASDNav({
  routes,
  color = 'white',
  gap = '2rem',
  background = 'transparent',
  isLandingPage,
}: IWASDNavProps): JSX.Element {
  return (
    <WASDGrid gap={gap}>
      {routes.map(
        (route: IRouteObject, i: number): JSX.Element => (
          <Tween
            key={i}
            from={
              isLandingPage
                ? { boxShadow: `0 0 0 32rem inset white`, opacity: 0, scale: 0 }
                : {}
            }
            duration={1}
            delay={i * 0.25}
            ease={'bounce'}
          >
            <StyledLinkOrb
              key={i}
              to={route.path}
              activeClassName="active"
              partiallyActive
              color={color}
              bordercolor={background === 'transparent' ? color : background}
              background={background}
              area={route.boundKeys[0].toLocaleLowerCase()}
            >
              <StyledLinkOrbKey>{`0${route.boundKeys[1]}`}</StyledLinkOrbKey>
              <StyledLinkOrbTitle>{route.title}</StyledLinkOrbTitle>
              <StyledLinkOrbKey>
                {route.boundKeys[0].toLocaleUpperCase()}
              </StyledLinkOrbKey>
            </StyledLinkOrb>
          </Tween>
        )
      )}
    </WASDGrid>
  )
}
