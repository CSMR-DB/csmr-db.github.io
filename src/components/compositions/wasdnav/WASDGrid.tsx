import styled, {
  StyledComponent,
  css,
  FlattenSimpleInterpolation,
  DefaultTheme,
} from 'styled-components'

export interface IWASDGridProps {
  theme?: DefaultTheme
  gap: string
}

export const WASDGrid: StyledComponent<
  'nav',
  any,
  IWASDGridProps,
  never
> = styled.nav`
  margin: 0 auto;
  display: grid;
  gap: 2rem;
  justify-content: space-between;
  align-items: center;
  grid-template-areas:
    '. w .'
    'a s d';
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;

  & > * {
    justify-self: center;
  }

  ${({ theme, gap }: IWASDGridProps): FlattenSimpleInterpolation => css`
    @media ${theme!.breakpoints.min.smartphone} {
      gap: ${gap};
    }
  `}
`
