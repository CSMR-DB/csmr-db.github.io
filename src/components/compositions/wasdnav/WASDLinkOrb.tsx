import styled, {
  StyledComponent,
  css,
  FlattenSimpleInterpolation,
  DefaultTheme,
} from 'styled-components'
import { GridAreaProperty } from 'csstype'

import { TestAttrs } from '../../../types/jest.types'

import { Link } from 'gatsby'

interface IWASDLinkOrbProps {
  theme?: DefaultTheme
  $area: GridAreaProperty
  $_testid: string
}

export const WASDLinkOrb: StyledComponent<
  typeof Link,
  any,
  IWASDLinkOrbProps,
  never
> = styled(Link).attrs(
  (props: IWASDLinkOrbProps): TestAttrs => ({ 'data-testid': props.$_testid })
)`
  width: 4rem;
  height: 4rem;
  margin: 0 auto;
  text-align: center;
  text-decoration: none;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  border-radius: 256rem;

  ${({ theme, $area }: IWASDLinkOrbProps): FlattenSimpleInterpolation => css`
    background: ${theme!.palette.first.lens};
    color: ${theme!.palette.light.normal};
    grid-area: ${$area};

    @media ${theme!.breakpoints.min.smartphone} {
      transition: background 200ms ease-in-out;
      width: 8rem;
      height: 8rem;

      &:hover {
        background: ${theme!.palette.first.hover};
        color: ${theme!.palette.light.hover};
      }
    }

    &.active {
      background: ${theme!.palette.first.active};
      color: ${theme!.palette.light.active};
    }
  `}
`
