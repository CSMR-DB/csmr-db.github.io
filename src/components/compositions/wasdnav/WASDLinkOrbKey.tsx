import styled, {
  StyledComponent,
  FlattenSimpleInterpolation,
  css,
  DefaultTheme,
} from 'styled-components'

export interface IWASDLinkOrbKeyProps {
  theme?: DefaultTheme
}

export const WASDLinkOrbKey: StyledComponent<
  'span',
  any,
  {},
  never
> = styled.span`
  display: none;
  border-radius: 4px;
  width: 2rem;
  height: 2rem;
  margin: 0 auto;
  line-height: 2rem;

  ${({ theme }: IWASDLinkOrbKeyProps): FlattenSimpleInterpolation => css`
    border: 1px solid ${theme!.palette.light.normal};

    @media ${theme!.breakpoints.min.smartphone} {
      display: initial;
    }
  `}
`
