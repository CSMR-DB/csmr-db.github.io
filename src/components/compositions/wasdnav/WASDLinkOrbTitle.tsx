import styled, { StyledComponent } from 'styled-components'

import { theme } from '../../../data/theme'

export const WASDLinkOrbTitle: StyledComponent<
  'h1',
  any,
  {},
  never
> = styled.h1`
  line-height: inherit;
  margin: 0;

  @media ${theme.breakpoints.max.smartphone} {
    font-size: 0.75rem;
  }
`
