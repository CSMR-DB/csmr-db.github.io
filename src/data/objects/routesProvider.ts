export interface IRouteObject {
  path: string
  boundKeys: [string, number]
  title: string
}

const ROUTES: IRouteObject[] = [
  {
    path: '/about',
    boundKeys: ['w', 0],
    title: 'about',
  },
  {
    path: '/contact',
    boundKeys: ['a', 1],
    title: 'contact',
  },
  {
    path: '/skillset',
    boundKeys: ['s', 2],
    title: 'skillset',
  },
  {
    path: '/projects',
    boundKeys: ['d', 3],
    title: 'projects',
  },
]

export function routesProvider(): IRouteObject[] {
  return ROUTES
}
