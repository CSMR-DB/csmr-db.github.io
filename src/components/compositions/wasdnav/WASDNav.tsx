import React from 'react'
import { Tween } from 'react-gsap'

import { RouteObject } from '../../../data/objects/routesProvider'
import { capitalizeString } from '../../../utils/textTransformer'

import { WASDGrid } from './WASDGrid'
import { WASDLinkOrb } from './WASDLinkOrb'
import { WASDLinkOrbTitle } from './WASDLinkOrbTitle'
import { WASDLinkOrbKey } from './WASDLinkOrbKey'

interface IWASDNavProps {
  routes: RouteObject[]
  isLandingPage?: boolean
}

export function WASDNav({
  routes,
  isLandingPage = false,
}: IWASDNavProps): JSX.Element {
  return (
    <WASDGrid>
      {routes.map(
        (route: RouteObject, i: number): JSX.Element => (
          <Tween
            key={i}
            from={isLandingPage && { opacity: 0, scale: 0 }}
            duration={1}
            delay={route.boundKeys[0] === 's' ? 0.75 : 1}
            ease={'back'}
          >
            <WASDLinkOrb
              key={i}
              to={route.path}
              activeClassName="active"
              partiallyActive
              $area={route.boundKeys[0].toLocaleLowerCase()}
              $_testid={`NavTo${capitalizeString(route.title)}`}
            >
              <WASDLinkOrbKey>{`0${route.boundKeys[1]}`}</WASDLinkOrbKey>
              <WASDLinkOrbTitle>{route.title}</WASDLinkOrbTitle>
              <WASDLinkOrbKey>
                {route.boundKeys[0].toLocaleUpperCase()}
              </WASDLinkOrbKey>
            </WASDLinkOrb>
          </Tween>
        )
      )}
    </WASDGrid>
  )
}
