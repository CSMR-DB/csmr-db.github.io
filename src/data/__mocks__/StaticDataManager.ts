import { ExperienceData } from './../graphql/experienceDataProvider'
import { DynamicImages } from './../graphql/dynamicImagesProvider'
import { DynamicIcons } from './../graphql/dynamicIconsProvider'
import { ProjectsData } from '../graphql/projectsDataProvider'
import { SkillsetData } from '../graphql/skillsetDataProvider'
import { GameWallpapers } from '../graphql/gameWallpapersProvider'
import { SeriesPosters } from '../graphql/seriesPostersProvider'
import {
  SiteMetadata,
  SkillsetAllMarkdownRemark,
  ImageSharpAllFiles,
  ProjectsAllMarkdownRemark,
  ProjectMarkdownEdge,
  SkillsetMarkdownEdge,
  ExperienceAllMarkdownRemark,
} from '../../types/graphql.types'
import { PhotographySquares } from '../graphql/photographySquaresProvider'
import { PhotographyData } from '../graphql/photographyDataProvider'
import { Tool } from '../objects/toolsProvider'
import { AStaticDataManager } from '../StaticDataManager'
import { RouteObject } from '../objects/routesProvider'
import { Course } from '../objects/coursesProvider'
import { Contact } from '../objects/contactsProvider'
import { DefaultTheme } from 'styled-components'
import { mockSVGs } from './tools.mocks'

const mockImageCollection: ImageSharpAllFiles = {
  edges: [
    {
      node: {
        name: 'Photo',
        relativePath: '/photo.jpg',
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: '',
            src: '/photo.jpg',
            srcSet: '/photo.jpg',
          },
        },
      },
    },
  ],
}

const mockProgrammingSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'React',
          excerpt: 'I love JSX',
          favorite: true,
          icon: '',
          skillColor: 'red',
          time: 1337,
        },
        generatedRoute: '/skillset/react',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Jest',
          excerpt: 'It Jest needs to be done',
          favorite: true,
          icon: '',
          skillColor: 'blue',
          time: 1336,
        },
        generatedRoute: '/skillset/jest',
      },
    },
  ],
}

const mockProgrammingProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Portfolio',
          excerpt:
            'Gatsby, TypeScript, React, GraphQL, GSAP, Styled-Components, Jest, Enzyme',
          favorite: true,
          tags: [
            'Gatsby',
            'TypeScript',
            'React',
            'GraphQL',
            'GSAP',
            'Styled-Components',
            'Jest',
            'Enzyme',
          ],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Loopy',
          excerpt: 'I made a fantastic hula-loop once',
          favorite: true,
          tags: ['Hula', 'Loop', 'Perfection'],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
  ],
}

const mockGraphicDesignProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Circle',
          excerpt: 'I drew a fantastic circle once',
          favorite: true,
          tags: ['Pencil', 'Shapes', 'Drawing'],
        },
        generatedRoute: '/projects/circle',
      },
    },
  ],
}

const mockGraphicDesignSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Affinity Designer',
          excerpt: 'It has no proper geometry operations',
          favorite: false,
          icon: '',
          skillColor: 'green',
          time: 420,
        },
        generatedRoute: '/skillset/affinity_designer',
      },
    },
  ],
}

const mockExperienceMdx: ExperienceAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          title: 'React',
          backgroundColor: 'blue',
          type: 'Reactor',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'JSX is just fantastic',
        },
        generatedRoute: '',
      },
    },
    {
      node: {
        frontmatter: {
          title: 'Jest',
          backgroundColor: 'red',
          type: 'Jester',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'Jest a test',
        },
        generatedRoute: '',
      },
    },
  ],
}

export class StaticDataManager implements AStaticDataManager {
  static get theme(): DefaultTheme {
    return {
      palette: {
        dark: {
          normal: '#000',
          hover: '#222',
          active: '#444',
          lens: '#666',
        },
        light: {
          normal: '#fff',
          hover: '#ddd',
          active: '#bbb',
          lens: '#999',
        },
        first: { normal: '#f00', hover: '#d00', active: '#b00', lens: '#900' },
        second: {
          normal: '#0f0',
          hover: '#0d0',
          active: '#0b0',
          lens: '#090',
        },
        third: { normal: '#00f', hover: '#00d', active: '#00b', lens: '#009' },
        filter: {
          normal: 'saturation(4)',
          hover: 'saturation(4)',
          active: 'saturation(4)',
        },
      },
      breakpoints: {
        min: {
          smartphone: 'only screen and (min-width: 600px)',
          tablet: 'only screen and (min-width: 960px)',
          desktop: 'only screen and (min-width: 1200px)',
          ultrawide: 'only screen and (min-width: 1920px)',
        },
        max: {
          smartphone: 'only screen and (max-width: 600px)',
          tablet: 'only screen and (max-width: 960px)',
          desktop: 'only screen and (max-width: 1200px)',
          ultrawide: 'only screen and (max-width: 1920px)',
        },
      },
      fonts: {
        headings: 'Arial',
        copy: 'serif',
      },
    }
  }

  static get siteMetadata(): SiteMetadata {
    return {
      author: 'Jester',
      description: 'The TestRunner',
      image: '/',
      title: 'Jest',
      url: 'https://www.jest.testrunner',
    }
  }

  static get contacts(): Contact[] {
    return [
      {
        background: 'red',
        href: 'https://mock.test.nowhere',
        iconPath: '',
        text: 'Mock Option',
      },
    ]
  }

  static get courses(): Course[] {
    return [
      {
        category: 'Development',
        description: 'Experimenting with vanilla JS & jQuery',
        name: 'WebProgramming',
      },
    ]
  }

  static get routes(): RouteObject[] {
    return [
      {
        path: '/route',
        title: 'Route',
        boundKeys: ['R', 1],
      },
    ]
  }

  static get tools(): Tool[] {
    return [
      {
        name: 'React',
        svg: mockSVGs().RandomSampleSVG,
        color: 'red',
      },
    ]
  }

  static get photographyData(): PhotographyData {
    return {
      photographyData: mockImageCollection,
    }
  }

  static get photographySquares(): PhotographySquares {
    return {
      photographySquares: mockImageCollection,
    }
  }

  static get seriesPosters(): SeriesPosters {
    return {
      seriesPosters: mockImageCollection,
    }
  }

  static get gameWallpapers(): GameWallpapers {
    return {
      gameWallpapers: mockImageCollection,
    }
  }

  static get skillsetData(): SkillsetData {
    return {
      all: { ...mockProgrammingSkillsetMdx, ...mockGraphicDesignSkillsetMdx },
      top: {
        edges: [
          ...mockProgrammingSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
          ...mockGraphicDesignSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingSkillsetMdx,
      allGraphicDesign: mockGraphicDesignSkillsetMdx,
    }
  }

  static get projectsData(): ProjectsData {
    return {
      all: { ...mockGraphicDesignProjectsMdx, ...mockProgrammingProjectsMdx },
      top: {
        edges: [
          ...mockGraphicDesignProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
          ...mockProgrammingProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingProjectsMdx,
      allGraphicDesign: mockGraphicDesignProjectsMdx,
    }
  }

  static get experienceData(): ExperienceData {
    return {
      experienceData: mockExperienceMdx,
    }
  }

  static get dynamicIcons(): DynamicIcons {
    return {
      dynamicIcons: mockImageCollection,
    }
  }

  static get dynamicImages(): DynamicImages {
    return {
      dynamicImages: mockImageCollection,
    }
  }
}
