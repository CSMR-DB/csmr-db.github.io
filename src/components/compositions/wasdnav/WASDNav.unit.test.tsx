import React, { Component } from 'react'
import { ThemeProvider } from 'styled-components'
import { create, ReactTestRenderer } from 'react-test-renderer'
import { mount, ReactWrapper } from 'enzyme'

import { StaticDataManager } from '../../../data/StaticDataManager'
import { RouteObject } from '../../../data/objects/routesProvider'

import { WASDNav } from './WASDNav'
import { WASDGrid } from './WASDGrid'
import { WASDLinkOrb } from './WASDLinkOrb'
import { WASDLinkOrbTitle } from './WASDLinkOrbTitle'
import { WASDLinkOrbKey } from './WASDLinkOrbKey'

describe('WASDNav', (): void => {
  it('should match snapshot', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <WASDNav routes={StaticDataManager.routes} isLandingPage={false} />
      </ThemeProvider>
    )

    expect(comp).toMatchSnapshot()
  })

  it('should render internally', (): void => {
    const comp: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>> = mount(
      <ThemeProvider theme={StaticDataManager.theme}>
        <WASDNav routes={StaticDataManager.routes} isLandingPage={false} />
      </ThemeProvider>
    )

    expect(comp.find(WASDGrid).length).toEqual(1)
    expect(comp.find(WASDLinkOrb).length).toEqual(
      StaticDataManager.routes.length
    )
    expect(comp.find(WASDLinkOrb).first().props().to).toEqual(
      StaticDataManager.routes[0].path
    )
    expect(comp.find(WASDLinkOrb).find(WASDLinkOrbTitle).length).toEqual(
      StaticDataManager.routes.length
    )
    expect(comp.find(WASDLinkOrb).first().find(WASDLinkOrbKey).length).toEqual(
      2
    )
    expect(comp.find(WASDLinkOrb).find(WASDLinkOrbKey).length).toEqual(
      StaticDataManager.routes.reduce(
        (acc: number, val: RouteObject): number => acc + val.boundKeys.length,
        0
      )
    )
  })
})
