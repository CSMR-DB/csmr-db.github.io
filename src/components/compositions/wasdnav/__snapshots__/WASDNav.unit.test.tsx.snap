// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WASDNav should match snapshot 1`] = `
.c0 {
  margin: 0 auto;
  display: grid;
  gap: 2rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-areas: '. w .' 'a s d';
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
}

.c0 > * {
  justify-self: center;
}

.c1 {
  width: 4rem;
  height: 4rem;
  margin: 0 auto;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  border-radius: 256rem;
  background: #900;
  color: #fff;
  grid-area: r;
}

.c1.active {
  background: #b00;
  color: #bbb;
}

.c3 {
  line-height: inherit;
  margin: 0;
  font-size: 0.75rem;
}

.c2 {
  display: none;
  border-radius: 4px;
  width: 2rem;
  height: 2rem;
  margin: 0 auto;
  line-height: 2rem;
  border: 1px solid #fff;
}

@media only screen and (min-width:600px) {
  .c0 {
    gap: 4rem;
  }
}

@media only screen and (min-width:600px) {
  .c1 {
    -webkit-transition: background 200ms ease-in-out;
    transition: background 200ms ease-in-out;
    width: 8rem;
    height: 8rem;
  }

  .c1:hover {
    background: #d00;
    color: #ddd;
  }
}

@media only screen and (min-width:600px) {
  .c3 {
    font-size: 1.25rem;
  }
}

@media only screen and (min-width:600px) {
  .c2 {
    display: initial;
  }
}

<nav
  className="c0"
>
  <a
    className="c1"
    href="/route"
  >
    <span
      className="c2"
    >
      01
    </span>
    <h1
      className="c3"
    >
      Route
    </h1>
    <span
      className="c2"
    >
      R
    </span>
  </a>
</nav>
`;
